### PR TITLE
Make sure the docs build with npm3 and npm2

### PR DESCRIPTION
**Pull request by zpao**

### CODE REVIEW

File: website/server/docgenHelpers.js [hunk 1]
```diff
@@ -1,5 +1,12 @@
 "use strict";
-var b = require('react-docgen/node_modules/recast').types.builders;
+var b;
+try {
+  // npm 3 probably
+  b = require('recast').types.builders;
```
Comment (ide) [inline website/server/docgenHelpers.js]: Shouldn't this work with npm 2 as well? Specifically, both npm 2 and npm 3 should set up node_modules in ways that are compatible with Node's require algo.

Comment (zpao) [inline website/server/docgenHelpers.js]: No. Node's algo doesn't descend to look for modules, only up.

```
.
├── node_modules
│   ├── react-docgen
│   │   └── node_modules
│   │       └── recast
│   └── something-else
└── server
    └── docgenHelpers.js
```

docgenHelpers will first look for `server/node_modules` (nope), then `$PWD/node_modules` (yay) then look for `recast` in that directory. It won't go looking recursively down from there there. Then it goes and continues walking up the filesystem (I think? I forget how this part works).

Comment (ide) [inline website/server/docgenHelpers.js]: Oh I see. Should package.json in `website` depend on recast then? (Or is this a weird case where recast is stateful and different packages need to access the same instance?)

Comment (zpao) [inline website/server/docgenHelpers.js]: That would probably be the better option :) I didn't actually look at any of the code here to figure out *why* it is like this, just made it run. #lazy

But now that I'm actually looking at the rest of this, we can just remove this whole require. `recast` is getting passed into the APIs below (you can see `b` get pulled off there so this one isn't even getting used).

**Comment by ide** [comment]
Merging this since the ObjC test is unrelated and is failing on other PRs. Thanks @zpao! I'll try upgrading the tests to use Node 5.... (fingers crossed!)

**Comment by zpao** [comment]
Noooo don't go to node 5 :( I think we should stick to using LTS versions for this sort of infra.

**Comment by ide** [comment]
People are going to be using Node 5 though. Move slow is for other companies!

**Comment by hswolff** [comment]
image

**Pull request merged**